src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/UploadEnvelopeDocumentsService.java

```diff
@@ -0,0 +1,219 @@
+            String message = String.format(
+                "Unable to get blob input stream. Container: %s, Blob: %s, Envelope ID: %s",
+                containerName,
+                zipFileName,
+                envelopeId
+            );
+
+            throw new UploadFailures(message, exception);
+        }
+    }
+
+    private ZipFileProcessingResult processInputStream(
+        BlobInputStream blobInputStream,
+        String containerName,
+        String zipFileName,
+        UUID envelopeId
+    ) {
+        try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
+            return zipFileProcessor.process(zis, zipFileName);
+        } catch (Exception exception) {
+            String message = String.format(
+                "Failed to process zip. File: %s, Container: %s, Envelope ID: %s",
+                zipFileName,
+                containerName,
+                envelopeId
+            );
+
+            createDocUploadFailureEvent(containerName, zipFileName, exception.getMessage(), envelopeId);
+
+            throw new UploadFailures(message, exception);
+        }
+    }
+
+    private void uploadParsedZipFileName(Envelope envelope, List<Pdf> pdfs) {
+        try {
+            documentProcessor.uploadPdfFiles(pdfs, envelope.getScannableItems());
+
+            log.info(
+                "Uploaded pdfs. File {}, Container: {}, Envelope ID: {}",
+                envelope.getZipFileName(),
+                envelope.getContainer(),
+                envelope.getId()
+            );
+        } catch (Exception exception) {
+            String message = String.format(
+                "Failed to upload PDF files to Document Management. File: %s, Container: %s, Envelope ID: %s",
+                envelope.getZipFileName(),
+                envelope.getContainer(),
+                envelope.getId()
+            );
+
+            envelopeProcessor.markAsUploadFailure(envelope);
+            createDocUploadFailureEvent(
+                envelope.getContainer(),
+                envelope.getZipFileName(),
+                exception.getMessage(),
+                envelope.getId()
+            );
+
+            throw new UploadFailures(message, exception);
+        }
+    }
+
+    private void createDocUploadFailureEvent(String containerName, String zipFileName, String reason, UUID envelopeId) {
+        envelopeProcessor.createEvent(
+            DOC_UPLOAD_FAILURE,
+            containerName,
+            zipFileName,
+            reason,
+            envelopeId
+        );
+    }
+
+    private void tryBreakLease(CloudBlockBlob blobClient) {
+        try {
+            blobClient.breakLease(0);
+        } catch (StorageException exception) {
+            // we will expire lease anyway. no need to escalate to error
+            log.warn("Failed to break the lease for {}", blobClient.getName(), exception);
+        }
+    }
+
+    private static class UploadFailures extends RuntimeException {
```

Comment:
Best to follow the existing naming convention for exceptions. How about `UploadFailureException` or `FailedUploadException` or `DocumentUploadException` or sth like this?

Reply:
I knew this will be a topic ;). Therefore left for suggestion in the review. Thanks @lgonczar, will do the first option. The reason I did such name is because I would like to keep it private only for internal use. Short and simple